OpenRA.Game/Support/ConditionExpression.cs

```diff
@@ -64,14 +64,15 @@ class Token
 			public TokenType Closes { get { return Type.Closes; } }
 			public bool Closer { get { return Type.Closes != null; } }
 
-			public Token(string symbol, int index, TokenType type)
+			protected Token(string symbol, int index, TokenType type)
 			{
 				Symbol = symbol;
 				Index = index;
 				Type = type;
 			}
 
-			public bool Is(TokenType type) { return object.ReferenceEquals(Type, type); }
+			public bool Is(TokenType type) { return ReferenceEquals(Type, type); }
+			public virtual void Evaluate(Stack<int> stack, IReadOnlyDictionary<string, int> variables) { }
```

Comment:
These evaluations happen many thousands of times per tick.  Have you checked for performance regressions?